BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/FileDiagnosticsTelemetryModule.cs

```diff
@@ -5,32 +5,37 @@
     using System;
     using System.Diagnostics;
     using System.Diagnostics.Tracing;
-    using System.Globalization;
     using System.IO;
+    using System.Runtime.CompilerServices;
     using System.Security;
     using System.Threading;
+
+    using Microsoft.ApplicationInsights.Common.Extensions;
     using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.FileDiagnosticsModule;
 
+    using static System.FormattableString;
+
     /// <summary>
     /// Diagnostics telemetry module for azure web sites.
     /// </summary>
     public class FileDiagnosticsTelemetryModule : IDisposable, ITelemetryModule
     {
+        private readonly TraceSourceForEventSource traceSource = new TraceSourceForEventSource(EventLevel.Error);
+        private readonly DefaultTraceListener listener = new DefaultTraceListener();
+
         private string windowsIdentityName;
-        
         private string logFileName;
         private string logFilePath;
 
-        private TraceSourceForEventSource traceSource = new TraceSourceForEventSource(EventLevel.Error);
-        private DefaultTraceListener listener = new DefaultTraceListener();
-
         /// <summary>
         /// Initializes a new instance of the <see cref="FileDiagnosticsTelemetryModule" /> class.
         /// </summary>
         public FileDiagnosticsTelemetryModule()
         {
             this.logFilePath = Environment.ExpandEnvironmentVariables("%TEMP%");
-            this.logFileName = "ApplicationInsightsLog_" + Process.GetCurrentProcess().Id.ToString(CultureInfo.InvariantCulture) + ".txt";
+
+            var process = Process.GetCurrentProcess();
+            this.logFileName = Invariant($"ApplicationInsightsLog_{DateTime.UtcNow.ToInvariantString("yyyyMMdd_HHmmss")}_{process.ProcessName}_{process.Id}.txt");
```

Comment:
Due to usage to utcnow, we anyway get a unique file name, so do we still need pid at all?

Reply:
yes because one machine may have multiple sites/apps running with the SDK